test/unit/snippets.cpp

```diff
@@ -931,6 +931,17 @@ struct CharSet
 };
 //]
 
+//[snippet_mutable_string_exemplar
+struct MutableString
+{
+    template< class InputIt >
+    MutableString& assign( InputIt first, InputIt last );
```

Comment:
You don't need the requirement that the function return `T&`

Reply:
I replaced `MutableString&` with `void`. But that would imply the requirement is `void`. Both are wrong, but `MutableString&` is more common although void might be safer (? not sure).